cmd/injector.go

```diff
@@ -49,11 +51,92 @@ func (e *FieldNotMatched) Is(err error) (matched bool) {
 	return
 }
 
+// TypeError used to report resource field cast error.
+type TypeError struct {
+	Field  *Field
+	Reason string
+	Object any
+}
+
+func (e *TypeError) Error() (s string) {
+	return fmt.Sprintf(
+		"Resource injector: cast failed. field=%s type=%s reason=%s, object:%v",
+		e.Field.Name,
+		e.Field.Type,
+		e.Reason,
+		e.Object)
+}
+
+func (e *TypeError) Is(err error) (matched bool) {
+	var inst *TypeError
+	matched = errors.As(err, &inst)
+	return
+}
+
 // Field injection specification.
 type Field struct {
-	Name string `json:"name"`
-	Path string `json:"path"`
-	Key  string `json:"key"`
+	Name    string `json:"name"`
+	Path    string `json:"path"`
+	Key     string `json:"key"`
+	Type    string `json:"type"`
+	Default any    `json:"default"`
+}
+
+// cast returns object cast as defined by Field.Type.
+func (f *Field) cast(object any) (cast any, err error) {
+	cast = object
+	if f.Type == "" {
+		return
+	}
+	defer func() {
+		if err != nil {
+			err = &TypeError{
+				Field:  f,
+				Reason: err.Error(),
+				Object: object,
+			}
+		}
+	}()
+	switch strings.ToLower(f.Type) {
+	case "string":
+		cast = fmt.Sprintf("%v", object)
+	case "integer":
+		switch x := object.(type) {
+		case int,
+			int8,
+			int16,
+			int32,
+			int64:
+			cast = x
+		case bool:
+			cast = 0
+			if x {
+				cast = 1
+			}
+		case string:
+			cast, err = strconv.Atoi(x)
+		default:
+			err = errors.New("expected: int|boolean|string")
+		}
+	case "boolean":
+		switch x := object.(type) {
+		case bool:
+			cast = x
+		case int,
+			int8,
+			int16,
+			int32,
+			int64:
+			cast = x != 0
+		case string:
+			cast, err = strconv.ParseBool(x)
+		default:
+			err = errors.New("expected: int|boolean|string")
+		}
+	default:
+		err = errors.New("expected: integer|boolean")
```

Comment:
why is this only `integer|boolean`?

Reply:
I added _string_ afterwards and forgot to add here.  Fixed.